tcl/pd_canvaszoom.tcl

```diff
@@ -0,0 +1,333 @@
+package provide pd_canvaszoom 0.1
+
+namespace eval ::pd_canvaszoom:: {
+    namespace export zoominit
+    namespace export canvasxy
+    namespace export scalescript
+}
+
+proc ::pd_canvaszoom::zoominit {mytoplevel {zfact {1.1}}} {
+    set c [tkcanvas_name $mytoplevel]
+    # save zoom state in a global variable with the same name as the canvas handle
+    upvar #0 $c data
+    set data(zdepth) 1.0
+    set data(oldzdepth) 1.0
+    set data(idle) {}
+
+    # add mousewheel bindings to canvas
+    if {$::windowingsystem eq "x11"} {
+        bind all <Control-Button-4> \
+            {event generate [focus -displayof %W] <Control-MouseWheel> -delta  1}
+        bind all <Control-Button-5> \
+            {event generate [focus -displayof %W] <Control-MouseWheel> -delta -1}
+    }
+    bind $c <Control-MouseWheel> "if {%D > 0} {zoom $c $zfact} else {zoom $c [expr {1.0/$zfact}]}"
+
+    # add button-2 bindings to scroll the canvas
+    bind $c <ButtonPress-2> {+ %W scan mark %x %y}
+    bind $c <B2-Motion> {+ %W scan dragto %x %y 1}
+}
```

Comment:
to be honest i think i would prefer it, if continuous zoom wouldn't use the mousewheel, but instead just have <kbd>Ctrl</kbd>+<kbd>+</kbd> resp <kbd>Ctrl</kbd>+<kbd>-</kbd> (this is already the established shortcut for zoom-in/zoom-out; and the mousewheel is not very comfortable in a laptop (without a realy mouse))

Reply:
I prefer mousewheel to scroll the canvas
any of that can also be done in a tcl plugin probably though..

Reply:
I find  <kbd>Ctrl</kbd>+<kbd>mousewheel</kbd> very convenient, and it's a very common shortcut.

I'm all for using <kbd>Ctrl</kbd>+<kbd>+</kbd>/<kbd>Ctrl</kbd>+<kbd>-</kbd> shortcuts together, but obviously that would mean getting rid of the old zoom functionality.

Reply:
I reassigned the Ctrl+ and Ctrl- shortcuts, as well as Edit/Zoom* menu entries.
So now the legacy zoom function isn't reachable anymore...